services/processor/processor/lib/utils.py

```diff
@@ -0,0 +1,570 @@
+            # shotgun_api3.shotgun.Fault: API schema_field_read()
+            missing_attrs.append(ayon_attr)
+
+    return missing_attrs
+
+
+def get_sg_project_by_id(
+    sg_session: shotgun_api3.Shotgun,
+    project_id: int,
+    extra_fields: Optional[list] = None
+) -> dict:
+    """ Find a project in Shotgrid by its id.
+
+    Args:
+        sg_session (shotgun_api3.Shotgun): Shotgun Session object.
+        project_id (int): The project ID to look for.
+        extra_fields (Optional[list]): List of optional fields to query.
+    Returns:
+        sg_project (dict): Shotgrid Project dict.
+     """
+    common_fields = list(SG_COMMON_ENTITY_FIELDS)
+
+    if extra_fields:
+        common_fields.extend(extra_fields)
+
+    sg_project = sg_session.find_one(
+        "Project",
+        [["id", "is", project_id]],
+        fields=common_fields,
+    )
+
+    if not sg_project:
+        raise ValueError(f"Unable to find project {project_id} in ShotGrid.")
+
+    return sg_project
+
+
+def get_sg_project_by_name(
+    sg_session: shotgun_api3.Shotgun,
+    project_name: str,
+) -> dict:
+    """ Find a project in Shotgrid by its name.
+
+    Args:
+        sg_session (shotgun_api3.Shotgun): Shotgun Session object.
+        project_name (str): The project name to look for.
+    Returns:
+        sg_project (dict): Shotgrid Project dict.
+     """
+    sg_project = sg_session.find_one(
+        "Project",
+        [["name", "is", project_name]],
+        fields=["id", "code", "name", "sg_status"],
+    )
+
+    if not sg_project:
+        raise ValueError(f"Unable to find project {project_name} in ShotGrid.")
+
+    return sg_project
+
+
+def get_sg_project_enabled_entities(
+    sg_session: shotgun_api3.Shotgun,
+    sg_project: dict
+) -> list:
+    """Function to get all enabled entities in a project.
+
+    Shotgrid allows a lot of flexibility when it comes to hierarchies, here we
+    find all the enabled entity type (Shots, Sequence, etc) in a specific
+    project and provide the configured field that points to the parent entity.
+
+    Args:
+        sg_session (shotgun_api3.Shotgun): Shotgun Session object.
+        project_name (str): The project name to look for.
+
+    Returns:
+        project_entities (list[tuple(entity type, parent field)]): List of
+            enabled entities names and their respective parent field.
+    """
+    sg_project = sg_session.find_one(
+        "Project",
+        filters=[["id", "is", sg_project["id"]]],
+        fields=["tracking_settings"]
+    )
+
+    if not sg_project:
+        logging.error(
+            f"Unable to find {sg_project} in the Shotgrid instance."
+        )
+        return []
+
+    sg_project_schema = sg_session.schema_entity_read(
+        project_entity=sg_project
+    )
+
+    project_navigation = sg_project["tracking_settings"]["navchains"]
+    project_navigation["Task"] = "entity"
+
+    project_entities = []
+
+    for entity in AYON_SHOTGRID_ENTITY_MAP:
```

Comment:
I would recommend to change the variable to `entity_type`.

Reply:
I agree, the problem is that SG entities are just dicts... I guess I could add `_dict` everywhere

Reply:
I would not complain if it would be a dictionary, but this is string which holds entity type.

Reply:
Aha, yeah, I got that, pushing a change to clarify this